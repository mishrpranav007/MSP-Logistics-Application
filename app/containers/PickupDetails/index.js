import React, { Component } from 'react';
import {
  StyleSheet,
  ImageBackground,
  View,
  Text,
  Image,
  TouchableOpacity,
  Platform
} from 'react-native';
import { injectIntl } from 'react-intl';
import pickupdetailsimage from 'app/assets/images/Pickupdetails.png';
import PropTypes from 'prop-types';
import Mask from 'react-native-mask';
import addimage from 'app/assets/images/addimage.png';
import ImagePicker from 'react-native-image-picker';
import { Dropdown } from 'react-native-material-dropdown';
import Style from '../../style/index';

class PickupDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      photo: null
    };
  }

  handleChoosePhoto = () => {
    const options = {
      noData: true
    };
    ImagePicker.launchImageLibrary(options, response => {
      if (response.uri) {
        this.setState({ photo: response });
      }
    });
  };

  render() {
    const { photo } = this.state;
    const data = [
      { value: 'Bag damaged' },
      { value: 'Bag dirty' },
      { value: 'Icepack missing' },
      { value: 'Icepack missed' }
    ];
    return (
      <ImageBackground
        resizeMode="cover"
        source={pickupdetailsimage}
        style={styles.backgroundContainer}
      >
        <View style={styles.textContainer}>
          <Text style={styles.textView}>Jhon Smith</Text>
          <View style={styles.textUnderlineView}></View>
          <Text style={styles.deliveryDetailsText}>Pickup Details</Text>
          <View style={styles.borderLine}></View>
          <View style={styles.rectangleShapeView}>
            <Text style={styles.customerText}>Sarah Jane</Text>
            <Text style={styles.customernameText}>Bag no. MSP14523</Text>
          </View>

          <View style={styles.shapeSize}>
            <Mask shape="rounded">
              {photo && (
                <Image
                  style={styles.shapeSizeImage}
                  source={{ uri: photo.uri }}
                />
              )}

              {!photo && (
                <View>
                  <TouchableOpacity onPress={this.handleChoosePhoto}>
                    <Image
                      source={addimage}
                      style={styles.addimageview}
                    ></Image>
                  </TouchableOpacity>

                  <Text style={styles.maskedText}>Add Bag photo</Text>
                </View>
              )}
            </Mask>
          </View>

          <View style={styles.checkboxRectangleView}>
            <Text style={styles.selectText}>Select state of bag</Text>
            <View style={styles.dropdownContainer}>
              <Dropdown data={data} style={styles.dropdownView} />
            </View>
          </View>

          <View>
            <TouchableOpacity style={styles.markpickupButton}>
              <Text style={styles.text}>
                {this.props.intl.formatMessage({ id: 'mark_picked' })}
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </ImageBackground>
    );
  }
}
const styles = StyleSheet.create({
  backgroundContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    marginTop: Style.em(5),
    marginLeft: Style.em(1)
  },
  textView: {
    color: Style.WHITE_COLOR,
    height: Style.em(0.813),
    marginRight: Style.em(2)
  },
  textUnderlineView: {
    height: Style.em(0),
    width: Style.em(4.1),
    borderTopColor: Style.WHITE_COLOR,
    borderTopWidth: 1,
    marginTop: Style.em(0.3),
    marginRight: Style.em(2)
  },
  deliveryDetailsText: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1.25),
    marginTop: Style.em(0.3),
    marginLeft: Style.em(1),
    marginRight: Style.em(13)
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  rectangleShapeView: {
    width: Style.em(19.5),
    height: Style.em(4.5),
    marginTop: Style.em(1.3),
    marginRight: Style.em(1.5),
    borderRadius: 8,
    backgroundColor: Style.WHITE_COLOR
  },
  customerText: {
    fontSize: Style.em(1),
    marginLeft: Style.em(1),
    marginTop: Style.em(0.7),
    textDecorationColor: Style.GRAY_COLOR
  },
  customernameText: {
    fontSize: Style.em(1.65),
    marginLeft: Style.em(1),
    marginTop: Style.em(0.1),
    textDecorationColor: Style.GRAY_COLOR
  },
  borderLine: {
    borderBottomColor: Style.WHITE_COLOR,
    borderBottomWidth: 1
  },
  shapeSize: {
    justifyContent: 'flex-end',
    alignItems: 'center',
    height: Style.em(19.5),
    width: Style.em(19),
    borderRadius: Style.em(0.8),
    marginTop: Style.em(1.5),
    marginRight: Style.em(1.8),
    backgroundColor: Style.POWDER_COLOR
  },
  addimageview: {
    marginLeft: Style.em(1.2)
  },
  maskedText: {
    marginBottom: Style.em(1.2),
    fontSize: Style.em(1.2),
    textDecorationColor: Style.PRIMARY_COLOR,
    marginTop: Style.em(5.2)
  },
  iconView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    marginTop: Style.em(7),
    marginLeft: Style.em(1)
  },
  checkboxRectangleView: {
    width: Style.em(18),
    height: Style.em(2.1),
    marginTop: Style.em(1),
    marginRight: Style.em(1.98),
    backgroundColor: Style.WHITE_COLOR,
    borderRadius: Style.em(0.25)
  },
  selectText: {
    marginTop: Style.em(0.5),
    marginLeft: Style.em(0.7),
    textDecorationColor: Style.SELECT_COLOR
  },
  markpickupButton: {
    width: Style.em(20),
    height: Style.em(3.6),
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Style.PRIMARY_COLOR,
    marginTop: Platform.OS === 'ios' ? Style.em(1.5) : Style.em(2),
    marginRight: Style.em(1.3),
    borderRadius: Style.em(0.69),
    marginBottom: Platform.OS === 'ios' ? Style.em(3.2) : Style.em(5),
    borderTopColor: Style.SECONDARY_COLOR,
    zIndex: 100,
    shadowColor: Style.WHITE_COLOR,
    shadowOffset: {
      width: 0,
      height: 10
    },
    shadowOpacity: 0.35,
    shadowRadius: 13.16,

    elevation: 20
  },
  text: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1.12),
    textAlign: 'center'
  },
  shapeSizeImage: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    width: Style.em(19.5),
    height: Style.em(19)
  },
  dropdownContainer: {
    flex: 1
  },
  dropdownView: {
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: Style.em(0.9),
    width: Style.em(10),
    height: Style.em(5),
    marginBottom: Style.em(10)
  }
});
PickupDetails.propTypes = {
  intl: PropTypes.object
};
export default injectIntl(PickupDetails);
