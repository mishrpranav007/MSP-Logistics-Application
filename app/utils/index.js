import { Platform, Dimensions } from 'react-native';

export const isAndroid = () => Platform.OS !== 'ios';
export const getScreenWidth = () => Dimensions.get('window').width;
export const getScreenHeight = () => Dimensions.get('window').height;

export const getPlatformString = () => (isAndroid() ? 'android' : 'ios');
export const getFontFamily = font =>
  isAndroid() ? `Futura-${font}` : `FuturaStd-${font}`;

export const isScreenSmall = () => getScreenHeight() <= 550;
