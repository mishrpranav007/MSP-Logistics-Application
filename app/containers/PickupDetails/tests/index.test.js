import React from 'react';
import { renderWithIntl } from 'app/utils/testUtils';
import PickupDetails from '../index';

describe('<PickupDetails />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<PickupDetails />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 button component inside the pickupdetails screen', () => {
    const { getAllByTestId } = renderWithIntl(<PickupDetails />);
    expect(getAllByTestId('pick-up-details-button').length).toBe(1);
  });
});
