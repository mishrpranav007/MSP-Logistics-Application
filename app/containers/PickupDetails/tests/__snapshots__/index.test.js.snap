// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PickupDetails /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/images/Pickupdetails.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 25.142857142857142,
            "marginTop": 125.71428571428571,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "height": 25.142857142857142,
              "marginRight": 50.285714285714285,
              "marginTop": 7.542857142857143,
            }
          }
        >
          Jhon Smith
        </Text>
        <View
          style={
            Object {
              "borderTopColor": "rgb(255,255,255)",
              "borderTopWidth": 1,
              "height": 0,
              "marginRight": 50.285714285714285,
              "marginTop": 7.542857142857143,
              "width": 103.08571428571427,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 28.160000000000004,
              "marginLeft": 10.057142857142857,
              "marginRight": 336.9142857142857,
              "marginTop": 7.542857142857143,
            }
          }
        >
          Delivery Details
        </Text>
        <View
          style={
            Object {
              "borderBottomColor": "rgb(255,255,255)",
              "borderBottomWidth": 1,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "rgb(255,255,255)",
              "borderRadius": 8,
              "height": 113.14285714285714,
              "marginRight": 35.199999999999996,
              "marginTop": 27.65714285714286,
              "width": 490.2857142857143,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 25.142857142857142,
                "marginLeft": 25.142857142857142,
                "marginTop": 17.599999999999998,
                "textDecorationColor": "rgb(109,109,109)",
              }
            }
          >
             
            Sarah Jane
          </Text>
          <Text
            style={
              Object {
                "fontSize": 41.48571428571428,
                "marginLeft": 25.142857142857142,
                "marginTop": 2.5142857142857142,
                "textDecorationColor": "rgb(109,109,109)",
              }
            }
          >
            Bag no. MSP14523
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(200,247,254)",
              "borderRadius": 20.114285714285714,
              "height": 490.2857142857143,
              "justifyContent": "flex-end",
              "marginRight": 40.22857142857143,
              "marginTop": 37.714285714285715,
              "width": 477.7142857142857,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 0,
                  "overflow": "hidden",
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 8,
                },
              ]
            }
          >
            <View>
              <TouchableOpacity
                activeOpacity={0.2}
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../app/assets/images/addimage.png",
                    }
                  }
                  style={
                    Object {
                      "marginLeft": 30.17142857142857,
                    }
                  }
                />
              </TouchableOpacity>
              <Text
                style={
                  Object {
                    "fontSize": 30.17142857142857,
                    "marginBottom": 30.17142857142857,
                    "marginTop": 130.74285714285713,
                    "textDecorationColor": "rgb(36,106,116)",
                  }
                }
              >
                 
                Add Bag photo
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgb(255,255,255)",
              "borderRadius": 6.285714285714286,
              "height": 52.8,
              "marginRight": 49.78285714285714,
              "marginTop": 25.142857142857142,
              "width": 452.57142857142856,
            }
          }
        >
          <Text
            style={
              Object {
                "marginLeft": 17.599999999999998,
                "marginTop": 12.571428571428571,
                "textDecorationColor": "rgba(0,0,0,0.65)",
              }
            }
          >
             
            Select state of bag
          </Text>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <TouchableWithoutFeedback
                disabled={false}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 4,
                    "right": 4,
                    "top": 6,
                  }
                }
              >
                <View
                  pointerEvents="box-only"
                >
                  <View
                    pointerEvents="none"
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                          "borderBottomWidth": 0.5,
                          "height": 63,
                          "paddingBottom": 8,
                          "paddingTop": 31,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "position": "absolute",
                            "top": 35,
                          }
                        }
                      >
                        <Text
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "rgba(0, 0, 0, 0.38)",
                              "fontSize": 16,
                            }
                          }
                        >
                          
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                          }
                        }
                      >
                        <TextInput
                          absoluteRTLLayout={false}
                          allowFontScaling={true}
                          autoCapitalize="sentences"
                          data={
                            Array [
                              Object {
                                "value": "Bag damaged",
                              },
                              Object {
                                "value": "Bag dirty",
                              },
                              Object {
                                "value": "Icepack missing",
                              },
                              Object {
                                "value": "Icepack missed",
                              },
                            ]
                          }
                          disableFullscreenUI={true}
                          dropdownMargins={
                            Object {
                              "max": 16,
                              "min": 8,
                            }
                          }
                          dropdownOffset={
                            Object {
                              "left": 0,
                              "top": 32,
                            }
                          }
                          editable={false}
                          itemColor="rgba(0, 0, 0, .54)"
                          itemCount={4}
                          itemPadding={8}
                          rejectResponderTermination={true}
                          rippleDuration={400}
                          selectionColor="rgb(0, 145, 234)"
                          shadeOpacity={0.12}
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "margin": 0,
                                "padding": 0,
                                "top": 2,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, .87)",
                                "fontSize": 16,
                                "height": 24,
                                "textAlign": "left",
                              },
                              Object {
                                "alignItems": "center",
                                "fontSize": 22.628571428571426,
                                "height": 125.71428571428571,
                                "justifyContent": "center",
                                "marginBottom": 251.42857142857142,
                                "width": 251.42857142857142,
                              },
                            ]
                          }
                          underlineColorAndroid="transparent"
                          useNativeDriver={false}
                          value=""
                        />
                        <View
                          style={
                            Object {
                              "alignSelf": "flex-start",
                              "justifyContent": "center",
                              "top": 2,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "height": 24,
                                "justifyContent": "center",
                                "width": 24,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "height": 6,
                                  "overflow": "hidden",
                                  "width": 12,
                                }
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 8,
                                      "transform": Array [
                                        Object {
                                          "translateY": -4,
                                        },
                                        Object {
                                          "rotate": "45deg",
                                        },
                                      ],
                                      "width": 8,
                                    },
                                    Object {
                                      "backgroundColor": "rgba(0, 0, 0, .38)",
                                    },
                                  ]
                                }
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "height": 8,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "bottom": 0,
                              "left": 0,
                              "paddingVertical": 4,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <Text
                            numberOfLines={1}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "rgb(213, 0, 0)",
                                "fontSize": 0,
                                "opacity": 0,
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "bottom": 0,
                              "left": 0,
                              "paddingVertical": 4,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <Text
                            numberOfLines={1}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "rgba(0, 0, 0, .38)",
                                "fontSize": 12,
                                "opacity": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <TouchableWithoutFeedback
                    disabled={false}
                  >
                    <View
                      pointerEvents="box-only"
                      rippleCentered={false}
                      rippleColor="rgba(0, 0, 0, .38)"
                      rippleDuration={400}
                      rippleFades={true}
                      rippleOpacity={0.54}
                      rippleSequential={true}
                      rippleSize={0}
                      style={
                        Object {
                          "height": 48,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 16,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "bottom": 0,
                              "left": 0,
                              "overflow": "hidden",
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            Object {
                              "borderRadius": 0,
                            },
                          ]
                        }
                      />
                    </View>
                  </TouchableWithoutFeedback>
                </View>
              </TouchableWithoutFeedback>
              <Modal
                hardwareAccelerated={false}
                supportedOrientations={
                  Array [
                    "portrait",
                    "portrait-upside-down",
                    "landscape",
                    "landscape-left",
                    "landscape-right",
                  ]
                }
                transparent={true}
                visible={false}
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "rgba(255, 255, 255, 1.0)",
                          "borderRadius": 2,
                          "position": "absolute",
                          "shadowColor": "rgba(0, 0, 0, 1.0)",
                          "shadowOffset": Object {
                            "height": 2,
                            "width": 0,
                          },
                          "shadowOpacity": 0.54,
                          "shadowRadius": 2,
                        },
                        Object {
                          "height": 176,
                          "left": undefined,
                          "top": undefined,
                          "transform": Array [
                            Object {
                              "translateY": -48,
                            },
                          ],
                          "width": undefined,
                        },
                        undefined,
                      ]
                    }
                  >
                    <ScrollView
                      contentContainerStyle={
                        Object {
                          "paddingVertical": 8,
                        }
                      }
                      data={
                        Array [
                          Object {
                            "value": "Bag damaged",
                          },
                          Object {
                            "value": "Bag dirty",
                          },
                          Object {
                            "value": "Icepack missing",
                          },
                          Object {
                            "value": "Icepack missed",
                          },
                        ]
                      }
                      disableVirtualization={false}
                      horizontal={false}
                      initialNumToRender={10}
                      maxToRenderPerBatch={10}
                      numColumns={1}
                      onEndReachedThreshold={2}
                      removeClippedSubviews={false}
                      scrollEnabled={false}
                      scrollEventThrottle={50}
                      stickyHeaderIndices={Array []}
                      style={
                        Object {
                          "borderRadius": 2,
                          "flex": 1,
                        }
                      }
                      updateCellsBatchingPeriod={50}
                      viewabilityConfigCallbackPairs={Array []}
                      windowSize={21}
                    >
                      <View
                        style={null}
                      >
                        <TouchableWithoutFeedback
                          disabled={false}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 4,
                              "right": 4,
                              "top": 6,
                            }
                          }
                        >
                          <View
                            focusAnimation={null}
                            focusAnimationDuration={225}
                            pointerEvents="box-only"
                            rippleCentered={false}
                            rippleColor="rgba(0, 0, 0, .38)"
                            rippleDuration={400}
                            rippleFades={true}
                            rippleOpacity={0.54}
                            rippleSequential={true}
                            rippleSize={0}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "borderRadius": 0,
                                "height": 40,
                                "justifyContent": "center",
                                "paddingLeft": undefined,
                                "paddingRight": undefined,
                                "paddingVertical": 8,
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              style={
                                Array [
                                  Object {
                                    "textAlign": "left",
                                  },
                                  undefined,
                                  Object {
                                    "color": "rgba(0, 0, 0, .87)",
                                    "fontSize": 16,
                                  },
                                ]
                              }
                            >
                              Bag damaged
                            </Text>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "rgba(0, 0, 0, .38)",
                                  "borderRadius": 0,
                                  "bottom": 0,
                                  "left": 0,
                                  "opacity": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                }
                              }
                            />
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "bottom": 0,
                                    "left": 0,
                                    "overflow": "hidden",
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  },
                                  Object {
                                    "borderRadius": 0,
                                  },
                                ]
                              }
                            />
                          </View>
                        </TouchableWithoutFeedback>
                      </View>
                      <View
                        style={null}
                      >
                        <TouchableWithoutFeedback
                          disabled={false}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 4,
                              "right": 4,
                              "top": 6,
                            }
                          }
                        >
                          <View
                            focusAnimation={null}
                            focusAnimationDuration={225}
                            pointerEvents="box-only"
                            rippleCentered={false}
                            rippleColor="rgba(0, 0, 0, .38)"
                            rippleDuration={400}
                            rippleFades={true}
                            rippleOpacity={0.54}
                            rippleSequential={true}
                            rippleSize={0}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "borderRadius": 0,
                                "height": 40,
                                "justifyContent": "center",
                                "paddingLeft": undefined,
                                "paddingRight": undefined,
                                "paddingVertical": 8,
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              style={
                                Array [
                                  Object {
                                    "textAlign": "left",
                                  },
                                  undefined,
                                  Object {
                                    "color": "rgba(0, 0, 0, .87)",
                                    "fontSize": 16,
                                  },
                                ]
                              }
                            >
                              Bag dirty
                            </Text>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "rgba(0, 0, 0, .38)",
                                  "borderRadius": 0,
                                  "bottom": 0,
                                  "left": 0,
                                  "opacity": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                }
                              }
                            />
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "bottom": 0,
                                    "left": 0,
                                    "overflow": "hidden",
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  },
                                  Object {
                                    "borderRadius": 0,
                                  },
                                ]
                              }
                            />
                          </View>
                        </TouchableWithoutFeedback>
                      </View>
                      <View
                        style={null}
                      >
                        <TouchableWithoutFeedback
                          disabled={false}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 4,
                              "right": 4,
                              "top": 6,
                            }
                          }
                        >
                          <View
                            focusAnimation={null}
                            focusAnimationDuration={225}
                            pointerEvents="box-only"
                            rippleCentered={false}
                            rippleColor="rgba(0, 0, 0, .38)"
                            rippleDuration={400}
                            rippleFades={true}
                            rippleOpacity={0.54}
                            rippleSequential={true}
                            rippleSize={0}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "borderRadius": 0,
                                "height": 40,
                                "justifyContent": "center",
                                "paddingLeft": undefined,
                                "paddingRight": undefined,
                                "paddingVertical": 8,
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              style={
                                Array [
                                  Object {
                                    "textAlign": "left",
                                  },
                                  undefined,
                                  Object {
                                    "color": "rgba(0, 0, 0, .87)",
                                    "fontSize": 16,
                                  },
                                ]
                              }
                            >
                              Icepack missing
                            </Text>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "rgba(0, 0, 0, .38)",
                                  "borderRadius": 0,
                                  "bottom": 0,
                                  "left": 0,
                                  "opacity": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                }
                              }
                            />
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "bottom": 0,
                                    "left": 0,
                                    "overflow": "hidden",
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  },
                                  Object {
                                    "borderRadius": 0,
                                  },
                                ]
                              }
                            />
                          </View>
                        </TouchableWithoutFeedback>
                      </View>
                      <View
                        style={null}
                      >
                        <TouchableWithoutFeedback
                          disabled={false}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 4,
                              "right": 4,
                              "top": 6,
                            }
                          }
                        >
                          <View
                            focusAnimation={null}
                            focusAnimationDuration={225}
                            pointerEvents="box-only"
                            rippleCentered={false}
                            rippleColor="rgba(0, 0, 0, .38)"
                            rippleDuration={400}
                            rippleFades={true}
                            rippleOpacity={0.54}
                            rippleSequential={true}
                            rippleSize={0}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "borderRadius": 0,
                                "height": 40,
                                "justifyContent": "center",
                                "paddingLeft": undefined,
                                "paddingRight": undefined,
                                "paddingVertical": 8,
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              style={
                                Array [
                                  Object {
                                    "textAlign": "left",
                                  },
                                  undefined,
                                  Object {
                                    "color": "rgba(0, 0, 0, .87)",
                                    "fontSize": 16,
                                  },
                                ]
                              }
                            >
                              Icepack missed
                            </Text>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "rgba(0, 0, 0, .38)",
                                  "borderRadius": 0,
                                  "bottom": 0,
                                  "left": 0,
                                  "opacity": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                }
                              }
                            />
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "bottom": 0,
                                    "left": 0,
                                    "overflow": "hidden",
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  },
                                  Object {
                                    "borderRadius": 0,
                                  },
                                ]
                              }
                            />
                          </View>
                        </TouchableWithoutFeedback>
                      </View>
                    </ScrollView>
                  </View>
                </View>
              </Modal>
            </View>
          </View>
        </View>
        <View>
          <TouchableOpacity
            activeOpacity={0.2}
            disabled={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(36,106,116)",
                "borderRadius": 17.348571428571425,
                "borderTopColor": "rgba(19,43,46,0.83)",
                "elevation": 20,
                "height": 80.45714285714286,
                "justifyContent": "center",
                "marginBottom": 80.45714285714286,
                "marginRight": 32.68571428571428,
                "marginTop": 37.714285714285715,
                "shadowColor": "rgb(255,255,255)",
                "shadowOffset": Object {
                  "height": 10,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 13.16,
                "width": 502.85714285714283,
                "zIndex": 100,
              }
            }
            testID="pick-up-details-button"
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 28.160000000000004,
                  "textAlign": "center",
                }
              }
            >
              Mark as picked
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </View>
</View>
`;
